libbeat/docs/generalconfig.asciidoc

```diff
@@ -103,3 +103,70 @@ processors in your config.
 
 Sets the maximum number of CPUs that can be executing simultaneously. The
 default is the number of logical CPUs available in the system.
+
+[float]
+[[configuration-internal-queue]]
+=== Configure the internal queue
+
+The `queue` namespace configures the queue type to be used.
+
+The Elastic Beats employ an internal queue for events to be published. The
+queue is responsible for buffering and combining events into batches, which can
+be consumed by the outputs. The outputs will use bulk operations to send a
+batch of events in one transaction.
+
+
+Example configuration:
+
+[source,yaml]
+------------------------------------------------------------------------------
+queue.mem:
+  events: 4096
+------------------------------------------------------------------------------
+
+==== Configure the Memory Qeueue
+[[configuration-internal-queue-memory]]
+
+The memory queue keeps all events in memory. By default no flush interval is
+configured. All events published to this queue will be directly consumed by the
+outputs. An output will consume up to the outputs `bulk_max_size` events at once.
+
+Only after an event is dropped or acknowledged by the output, a new slot will
```

Comment:
If the output is reading in batches, what do we mean by a slot? Does the slot open up before the batch is fully read? Maybe I'm not fully understanding the behavior here.